subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -43,18 +49,26 @@
 /**
  * Assembles the parts of a build script.
  */
+@SuppressWarnings("UnusedReturnValue")
 public class BuildScriptBuilder {
-
     private final BuildInitDsl dsl;
     private final String fileNameWithoutExtension;
     private boolean externalComments;
+    private final InsecureProtocolHandler insecureProtocolHandler;
 
     private final List<String> headerLines = new ArrayList<>();
-    private final TopLevelBlock block = new TopLevelBlock();
+    private final TopLevelBlock block;
 
     public BuildScriptBuilder(BuildInitDsl dsl, String fileNameWithoutExtension) {
+        this(dsl, fileNameWithoutExtension, InsecureProtocolOption.defaultOption);
```

Comment:
Looking at this some more, it seems like the code paths that go through this constructor are not supposed to have any insecure protocols used because the URLs should be coming from our code only. 

I wonder if we should have a slightly different implementation for `InsecureProtocolHandler` here where it always fails when it sees an insecure URL because that means our code to generate the project is wrong. The maven converter always goes through the other constructor and is user controlled.